### ISSUE TITLE
Some autotile layers not being imported

### ISSUE THREAD

**Issue by lenny991** [heygleeson/godot-ldtk-importer#21]
Hey. 

I have been trying to get this great asset to work for today now. Everything else works great, but larger tilesets do not. From the image I attached below, you can see the tilemaps (16x16) working in both LDTK and Godot, but the Big_Props tilemap (64x64) does not get shown. 

![image](https://github.com/user-attachments/assets/0bee40ba-d623-4f39-ad2a-6b745319c5ff)

A Tileset is created for it: 
![image](https://github.com/user-attachments/assets/1ba1388e-a77c-4df7-9b56-d75641f30d79)

I get the following errors when reimporting the .ldtk file:

```
core/variant/variant_utility.cpp:1112 - LDtk file version is newer than what is supported. Errors may occur.
  core/variant/variant_utility.cpp:1112 - TileSetSource creation failed!
  core/variant/variant_utility.cpp:1092 - TileSetAtlasSource missing
  core/variant/variant_utility.cpp:1092 - TileSetAtlasSource missing
  core/variant/variant_utility.cpp:1092 - TileSetAtlasSource missing
  core/variant/variant_utility.cpp:1092 - TileSetAtlasSource missing
  core/variant/variant_utility.cpp:1092 - TileSetAtlasSource missing
-- LDTK: Start Import --
LDTK VERSION OK
-- LDTK: Finished Import -- (108ms)
```

I also tried to do some debugging though the code was very complex for me :D
Seems that the layers.gd function create_layers() is not being called for the 64x64 tilemap, but I cannot really understand why.

Seems like a common thing to want bigger props such as trees in one's game. I'd be surprised if that was not possible, so I am guessing I am doing something wrong myself. Any help is very much appreciated, thanks!

For now I think I'll move to using entities for these larger props.

**Comment by lenny991** [comment]
This was really annoying me yesterday so I did some digging. Here's the reason: the layer import script is trying to look for 64x64 tiles in the 16x16 TileSet. This is because the grid size of the LDTK world is always 16x16. 

What I think would be the fix would be to use the grid size of the texture that is looked for to determine the TileSet to use, instead of the grid size of the tilemap. Because the grid size of an autotile layer cannot be changed, and will always be the same as the intgrid, even if the tiles are larger. 

Hope this helps, if someone will try to tackle this issue in the future.

**Comment by Joeyrp** [comment]
Hey! I have a potential fix for this issue I think. I'm not sure it's ready for a pull request yet because there are some changes I'm not really sure about. The example project seems work fine but I haven't done any other testing yet. I'm also still very new to LDtk so there may some obvious stuff that I'm missing. The commit with my fix is here though: https://github.com/Joeyrp/godot-ldtk-importer/commit/33d0cacac306f65872d934ebf0b31de408495e6e

The main thing I did was change the tileset Dictionary in Util.gd to make the tile set uid the key instead of the grid_size (the tileset res files are now saved with the uid instead of the grid_size as well). This allowed it to find the larger tiles on the auto-tile layer but they were offset incorrectly on the map. To fix this I changed an argument to use the tile_size instead of the grid_size and I marked that change with a big # HACK comment so it should be easy to see in the commit diff.

Without the hack solution the level looks like this: 
![incorrect_autotile_placement](https://github.com/user-attachments/assets/992faac5-b3d6-4ac8-a0d7-f8f6a63c6c9b)
The red boxes are the 64x64 tiles that I added for testing (the box at the top left is manually placed with a normal Tile layer and the rest are on the auto-tile layer). You can see the auto-tiled ones are off the map. With the hacky solution it seems to be fixed though:
![correct_autotile_placements](https://github.com/user-attachments/assets/419801c2-e3a1-4fce-99b6-e25216bea97f)

Hopefully this will help! I can try to make more adjustments/corrections if requested or open a pull request if it seems good enough.

**Comment by heygleeson** [comment]
Fascinating! I spent a while looking into how things worked in the importer in regards to this issue and it prompted me to rewrite TileSets. I didn't know LDtk supported using tilesets that don't match the layer grid size.

Rather than using @Joeyrp's solution, I found it was possible to still keep TilesetAtlasSources grouped by grid size, but 'mismatched' sources are copied into the TileSet if they are used in that grid size. This now behaves as you'd expect it to!

Thanks for the report! Fixed in [v2.0.1](https://github.com/heygleeson/godot-ldtk-importer/releases/tag/2.0.1)
